src/app/content/highlights/hooks/locationChange.ts

```diff
@@ -45,10 +45,18 @@ const hookBody = (services: MiddlewareAPI & AppServices) => async(action?: AnyAc
   const {book, page} = bookAndPage(state);
   const authenticated = user(state);
   const loaded = select.highlightsLoaded(state);
+  const recentlyLoadedFor = select.highlightsRecentlyLoadedFor(state);
 
   const pageFocusIn = action && action.type === getType(receivePageFocus) && action.payload;
 
-  if (!authenticated || !book || !page || typeof(window) === 'undefined' || (loaded && !pageFocusIn)) {
+  if (
+    !authenticated
+    || !book
+    || !page
+    || typeof(window) === 'undefined'
+    || (loaded && !pageFocusIn)
+    || (!pageFocusIn && recentlyLoadedFor === page.id)
```

Comment:
I might be wrong, but couldn't this be replaced by checking if `state.highlights` contains any highlight that has `sourceId === page.id`? I think it could be nice to avoid adding additional state if it's possible to retrive it from other locations.
 If I'm not missing anything i think the first highlight in `state.highlights` should have that source id so searching for it won't be hard on performance.

Reply:
I think this approach would not work if there are no highlights on a page

Adding new state like this is implemented in `requestBook` and `requestPage` which is called before `receiveBook` / `receivePage` to avoid re-fetching same book / page when it is already loading (`resolveContent.ts`) - it works a little bit different because this state is cleared when book / page is received and in our case, we just want to have information about which highlights are in the state

I was thinking about implementing additional action like `requestHighlights(page.id)` but this would have to be called before every `receiveHighlights` action + it would be a mistake if `receiveHighlights` would for some reason throw an error